### DISCUSSION TITLE
How to add an NTE after an OBX?

### DISCUSSION THREAD

**Discussion by bilago** [nHapiNET/nHapi#618] [Q&A] [answered]
I'm using the latest NHAPI in a .net 6 project and generating an `ORU_R01` message.
I wrote this method to add an `NTE` to an observation, but I cannot create an equivalent to put the `NTE` after the `OBX`.

```csharp
public static void CreateNote(ORU_R01_ORDER_OBSERVATION orderObservation, string? comment)
{
    var nte = orderObservation.AddNTE();
    nte.SetIDNTE.Value = orderObservation.NTERepetitionsUsed.ToString();
    //see https://hl7-definition.caristix.com/v2/HL7v2.5.1/Tables/0105 for list of valid sources
    nte.SourceOfComment.Value = "L";
    nte.GetComment(0).Value = comment;
}
```
        
The issue is that while the observation class has an `AddNTE()` method, there is none for the `OBX` record. Why is that missing if it is acceptable to have an `NTE` after an `OBX` according to the HL7 standard (unless I'm mistaken)

**Comment by milkshakeuk** [comment, marked as answer]
That is because you don't add NTE to OBX you can add NTE to `ORU_R01_ORDER_OBSERVATION` which is what you are referencing above and you can also add NTE segments to `ORU_R01_OBSERVATION` which is what you are after.

Here is an example of what I think you are trying to do.

```csharp
// your existing method
public static void CreateNote(ORU_R01_ORDER_OBSERVATION orderObservation, string? comment)
{
    var nte = orderObservation.AddNTE();
    nte.SetIDNTE.Value = orderObservation.NTERepetitionsUsed.ToString();
    //see https://hl7-definition.caristix.com/v2/HL7v2.5.1/Tables/0105 for list of valid sources
    nte.SourceOfComment.Value = "L";
    nte.GetComment(0).Value = comment;
}

// new method for observation notes
public static void CreateNote(ORU_R01_OBSERVATION observation, string? comment)
{
    var nte = observation.AddNTE();
    nte.SetIDNTE.Value = observation.NTERepetitionsUsed.ToString();
    //see https://hl7-definition.caristix.com/v2/HL7v2.5.1/Tables/0105 for list of valid sources
    nte.SourceOfComment.Value = "L";
    nte.GetComment(0).Value = comment;
}

...
// how to add the NTE after the OBX
oruR01.GetPATIENT_RESULT().GetORDER_OBSERVATION().GetOBSERVATION().CreateNote("note to add");
...
```

As a side note NET6 is out of support so is considered vulnerable to security exploits, I would consider updating to the current LTS at least which is NET8; should require minimal effort to do so.

**Reply by bilago** [reply]
Thank you, This was the answer, I feel silly for not realizing there was ORU_R01_ORDER_OBSERVATION AND ORU_RO1_OBSERVATION, which the latter was the one I was needing to add the NTE to. Once I made that change I'm now getting

OBR|1
OBX|1
NTE|1
NTE|2

versus before:

OBR|1
NTE|1
NTE|2
OBX|1


Regarding your supplemental comment regarding .net version - I agree with you but we're not approved for NET8 in prod yet.
